### PR TITLE
improved typing for admin client

### PR DESCRIPTION
**Pull request by 4xDMG**

### CODE REVIEW

File: src/admin/service.ts [hunk 1]
```diff
@@ -0,0 +1,357 @@
...
+    const rawAppUserJourney = await this.client.createAppUserJourney(appUserId, startDate);
+
+    return this.mapRawAppUserJourney(rawAppUserJourney);
+  };
+
+  createAppUserJourneyIndexEvent = async (
+    appUserId: string,
+    journeyId: number,
+    eventTypeSlug: string,
+    value: string,
```
Comment (elsimir) [inline src/admin/service.ts]: Suggestion: make a type representing a date time stored as a string to make it clear what data format is expected
```    
type ISO8601DateTime = string;
```

File: src/index.ts [hunk 2]
```diff
@@ -1,5 +1,5 @@
 import PathwaysClient, { IPathwaysClient, IOptions } from './client';
-import PathwaysAdminClient, { IPathwaysAdminClient, IAdminClientOptions } from './adminClient';
+import PathwaysAdminClient, { IPathwaysAdminClient, IAdminClientOptions } from './admin/client';
```
Comment (elsimir) [inline src/index.ts]: Missing re-export of PathwaysAdminService

File: src/admin/types.ts [hunk 3]
```diff
@@ -0,0 +1,324 @@
+export type MethodType = 'DELETE' | 'GET' | 'PATCH' | 'POST';
+export type WhatType = 'FEATURE_DOCUMENT' | 'FEATURE_FORM' | 'MESSAGE';
```
Comment (elsimir) [inline src/admin/types.ts]: These should use your constants.

File: src/admin/client.test.ts [hunk 4]
```diff
@@ -150,7 +150,7 @@ describe('Pathways client', () => {
     expect(f).toHaveBeenCalled();
     expect(f).toHaveBeenCalledWith(
       `${TEST_BASE_URL}appusers/`,
-      requestParameters('POST', {}, expect.objectContaining(body))
+      requestParameters('POST', {}, expect.objectContaining(body)),
```
[the remark this answers is not in the document]
Comment (4xDMG) [reply src/admin/client.test.ts]: Yep but I didn't get time to go back to testing during clean up. Do you want these updated as part of this?

File: src/admin/types.ts (hunk 3 above)
Comment (4xDMG) [reply src/admin/types.ts]: TypeScript didn't like it when I tried to use the constants. I tried for a bit to work it out this morning but moved on to the new stuff. I'll need your help for this one.

Comment (elsimir) [reply src/admin/types.ts]: No worries

File: src/admin/client.test.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (elsimir) [reply src/admin/client.test.ts]: I think its problematic to merge this with these tests being subtly broken. It'll be very easy in the future to assume that they work, and that passing tests when making other changes means that nothing has been broken. Better to either put the change in now, or to comment all of these and put a big block comment at the top explaining the problem.

File: src/admin/service.ts (hunk 1 above)
Comment (4xDMG) [reply src/admin/service.ts:159]: I think this comment was on the wrong line. I have added the type where I can see dates being using in the admin client.

**Pull request merged**